packages/api/src/car.js

```diff
@@ -273,6 +273,12 @@ async function carStat (carBlob) {
     if (blockSize > MAX_BLOCK_SIZE) {
       throw new InvalidCarError(`block too big: ${blockSize} > ${MAX_BLOCK_SIZE}`)
     }
+    if (block.cid.multihash.code !== sha256.code) {
+      throw new InvalidCarError(`block with unsupported hash function: ${block.cid.multihash.code} for ${block.cid.toString()}`)
+    }
+    if (!isValidBlock(block)) {
```

Comment:
yikes. this is async, so here we test that the return is truthy, and the non-null promise always returns true. Thank goodness for tests. Fixing now.